### ISSUE TITLE
groupComparison() Issue in ComparisonResult Data Frame

### ISSUE THREAD

**Issue by kkw5710** [MeenaChoi/MSstats#113]
Hi, I am still experiencing the same issue: the ComparisonResult data frame has only the Protein and Label columns with the actual (and correct) values, but every other column (i.e. log2FC, SE, Tvalue, DF, pvalue, adj.pvalue, etc.) only has NA for every observation for each protein. I have ensured that none of my group labels begin with digits by adding an 'X', and the issue still persists. Do you have other suggestions for how to resolve this?

**Comment by mstaniak** [comment]
Hi,
please see this gGroup thread: https://groups.google.com/u/2/g/msstats/c/RO_00B7Ou_k and run the same code. It will help us locating the error in model fitting

**Comment by kkw5710** [comment]
Thank you for your continued help. I just ran the following code and received the same error with both versions of the 'fitted_model'. 

<img width="640" alt="Screenshot 2024-03-04 113516" src="https://github.com/Vitek-Lab/MSstats/assets/74326179/dd3ce7f2-904b-4375-8177-a21718df1c6a">

**Comment by mstaniak** [comment]
Hi, 
now let's see the output of
```
    single_protein = MSstats:::.prepareSingleProteinForGC(single_protein)
    is_single_subject = MSstats:::.checkSingleSubject(single_protein)
    has_tech_reps = MSstats:::.checkTechReplicate(single_protein)
   
    fitted_model = MSstats:::.fitModelSingleProtein(single_protein, contrast_matrix,

                                              has_tech_reps, is_single_subject,
                                              repeated, groups, samples_info,
                                              save_fitted_models, has_imputed)
```

**Comment by mstaniak** [comment]
Hi,
thanks, I see, we need to dig a level deeper. Now please run:

```
 single_protein[, GROUP := factor(GROUP)]
 single_protein[, SUBJECT := factor(SUBJECT)]
    
 protein = unique(single_protein$Protein)
 n_groups = nlevels(single_protein$GROUP)
```

We need to see an example where n_groups is larger than 1.
For such an example, please show me the values of variables:
repeated, is_single_subject, has_tech_reps

**Comment by kkw5710** [comment]
Hi, here are the values of the variables you requested:
<img width="170" alt="Screenshot 2024-03-04 114643" src="https://github.com/Vitek-Lab/MSstats/assets/74326179/3921a223-8b04-421e-a282-624d439fc26e">

**Comment by mstaniak** [comment]
Then MSstats is trying to fit the following model:

```
lme4::lmer(ABUNDANCE ~ GROUP + (1|SUBJECT), data = single_protein)
```

please run this code and let me know what was the error

**Comment by kkw5710** [comment]
This is the error the code generates:

> lme4::lmer(ABUNDANCE ~ GROUP + (1|SUBJECT), data = single_protein)
Error in initializePtr() : 
  function 'cholmod_factor_ldetA' not provided by package 'Matrix'

**Comment by mstaniak** [comment]
OK, let's try this: please update the Matrix package, and then run:

```
install.packages("lme4", type = "source")
```

**Comment by kkw5710** [comment]
Ok, I have ensured that I am using the most recent Matrix package, and I have run the provided code.

**Comment by mstaniak** [comment]
Did re-installing lme4 from source fix the issue?

**Comment by kkw5710** [comment]
No, unfortunately, that did not fix the issue.

**Comment by mstaniak** [comment]
Does the same error persist? When running all the code for a single protein?

**Comment by kkw5710** [comment]
I still receive this error when running the code for a single protein:

> lme4::lmer(ABUNDANCE ~ GROUP + (1|SUBJECT), data = single_protein)
Error in initializePtr() : 
  function 'cholmod_factor_ldetA' not provided by package 'Matrix'

<img width="517" alt="Screenshot 2024-03-04 122510" src="https://github.com/Vitek-Lab/MSstats/assets/74326179/a9cec102-5f96-48dd-8d17-26aa3cfebc96">

**Comment by mstaniak** [comment]
what about just ```lme4::lmer(ABUNDANCE ~ GROUP + (1|SUBJECT), data = single_protein)```?

**Comment by mstaniak** [comment]
Error from fitModelSingleProtein only tells us model-fitting failed, unfortunately, we need to run all the code including ```lme4::lmer(ABUNDANCE ~ GROUP + (1|SUBJECT), data = single_protein)``` to see what the actual error is

**Comment by mstaniak** [comment]
Alternatively, you can try:

```
remove.packages("Matrix")
remove.packages("lme4")
install.packages("lme4", type = "source")
library(lme4)
```

it's a known problem that we have no control over:
https://stackoverflow.com/questions/77481539/error-in-initializeptr-function-cholmod-factor-ldeta-not-provided-by-pack

**Comment by kkw5710** [comment]
I have tried:
remove.packages("Matrix")
remove.packages("lme4")
install.packages("lme4", type = "source")
library(lme4)
I still, unfortunately, receive this error:
<img width="421" alt="Screenshot 2024-03-04 122843" src="https://github.com/Vitek-Lab/MSstats/assets/74326179/f6dbea83-0ad2-4e40-a3f9-4a3bc98fe0ec">

**Comment by mstaniak** [comment]
Did you refresh (=restart) R session?

**Comment by kkw5710** [comment]
I will have to get back to you. I am having difficulties now simply installing packages. Please do not close this issue, and I will respond to you as soon as this new matter has been resolved. Thank you for all of your help so far!

**Comment by kkw5710** [comment]
I truly want to thank you so much for all of your help! I finally managed to resolve the issue, and the groupComparison() output is back to normal once again! I had to:

1. Update the Write Permissions on the user account associated with my R and RStudio.
2. End any R and RStudio tasks on my Task Manager.
3. Completely delete my original folder files of the 'lme4' and 'Matrix' packages in my File Explorer.
4. Restart RStudio and install the 'Matrix' and 'lme4' packages again. 

So, it seemed as though the 'lme4' package was the cause of this issue. Thank you again so much!
